### PR TITLE
Add EIP associaton script

### PR DESCRIPTION
**Pull request by sevenmachines**
The change adds a script that can associate an EIP with an instance
based on pillar data listing ips that should be targeted. This is
part of the Covet IP pattern for maintaining EIP's on instances.

### CODE REVIEW

File: aws/map.jinja [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{% set aws = salt['grains.filter_by']({
+    'Default': {
+    	'eips': []
+    }
+}, grain='osfinger', merge=salt['pillar.get']('aws',{}), default='Default') %}
```
Comment (ltsampros) [inline aws/map.jinja:7]: @niallcreech Is this fore any eip that might have already been set to the the instance?

File: aws/files/autoeips.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            print ("retry: {}".format(retry) )
+            for eip in eips:
+                print ("Associating instance: {} with eip: {}".format(self.instance_id, eip.allocation_id))
+                success = eip.associate(instance_id=self.instance_id,
+                              allow_reassociation=False)
+                if success:
+                    return True
+            
+        return False
+    def get_unassociated_eips(self):
```
Comment (ltsampros) [inline aws/files/autoeips.py:80]: Minor pep8 thingie. I think it needs 2 new lines.

**Comment by sevenmachines** [comment]
@ltsampros sorry, the tag on this is wrong. this _really_ is a WIP. Theres a lot of fixups and production-ising to be done. Just wanted to get something up and running for testing the feasability of running this in production. Don't spend too much time delving into it, there'll be a lot of bugs and things to do, if you see possible problems in the concept though, that'd be good

**Comment by ltsampros** [comment]
@niallcreech ok. Currently, I don't get the grain thingie (see my first comment).

**Comment by sevenmachines** [comment]
@ltsampros It's just a pillar placeholder for having a better method of handling the case where no EIP addresses are specified in the user pillar. We can think of a good failure/default mode if the approach looks like a goer

File: aws/files/autoeips.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    
+    
+    def get_instance_metadata(self):
+        """
+        Prints a mapping of private ip addresses to private dns names
+        """
+        try:
+            instance_metadata = boto.utils.get_instance_metadata(timeout=5, num_retries=2)
+            return instance_metadata
+        except Exception as e:
```
Comment (lgarvey) [inline aws/files/autoeips.py]: It'd be cleaner if a single try/except happens in the update_association method, then all other exceptions can bubble up to that place.

**Comment by lgarvey** [comment]
Should the script be registering / unregistering with an ELB if it doesn't have a valid an EIP?

File: aws/files/autoeips.py [hunk 4]
```diff
@@ -101,9 +101,9 @@ def get_unassociated_eips(self):
 
             return unassociated_eips
 
-        except Exception as e:
+        except EC2ResponseError as e:
```
Comment (lgarvey) [inline aws/files/autoeips.py:102]: Think this needs importing. I removed the import as part of my pep8 cleanup!

Comment (sevenmachines) [reply aws/files/autoeips.py:102]: done

**Pull request merged**